File: web-forms/src/store/evaluation/cuidadores.js
import { defineStore } from 'pinia'
import { ref } from 'vue';

export const cuidadoresModel = {
    medico_solicitante: null,
    contato_medico: null,

    possui_cuidador: true,
    nome_cuidador: null,
    contato_cuidador: null,
    turno_cuidador: null,

    precisa_treinamento: false,
    obs_treinamento: null
}

export const useCuidadoresStore = defineStore('cuidadores', () => {

    const cuidadores = ref(JSON.parse(JSON.stringify(cuidadoresModel)))
    function reset() {
        cuidadores.value = JSON.parse(JSON.stringify(cuidadoresModel))
    }

    return {
        cuidadores,
        reset
    }
}, {
    persist: true
})

File: backend/internal/expansion/domain/cuidadores.go
package domain

type Cuidadores struct {
	MedicoSolicitante  string `json:"medico_solicitante"`
	ContatoMedico      string `json:"contato_medico"`
	PossuiCuidador     bool   `json:"possui_cuidador"`
	NomeCuidador       string `json:"nome_cuidador"`
	ContatoCuidador    string `json:"contato_cuidador"`
	TurnoCuidador      string `json:"turno_cuidador"`
	PrecisaTreinamento bool   `json:"precisa_treinamento"`
	ObsTreinamento     string `json:"obs_treinamento"`
}

func GetCuidadoresFormOptions() FormMetadata {
	return FormMetadata{
		"turno_cuidador": []FieldOption{
			{Label: "Noturno", Value: "turno_noturno"},
			{Label: "Diurno", Value: "turno_diurno"},
			{Label: "24h", Value: "turno_24h"},
		},
	}
}
